### PR TITLE
FIX: [longpress] fixes ghost repeat

### PR DESCRIPTION
**Pull request by koying**
fixes #16518

### CODE REVIEW

**Comment by MartijnKaijser** [comment]
jenkins build this please

File: xbmc/input/InputManager.cpp [hunk 1]
```diff
@@ -480,7 +480,11 @@ bool CInputManager::OnEvent(XBMC_Event& newEvent)
   case XBMC_KEYUP:
     m_Keyboard.ProcessKeyUp();
     if (m_LastKey.GetButtonCode() != KEY_INVALID && !(m_LastKey.GetButtonCode() & CKey::MODIFIER_LONG))
-      OnKey(m_LastKey);
+    {
+      CKey key = m_LastKey;
+      m_LastKey.Reset();  // OnKey is reentrant; need to do this before entering
+      OnKey(key);
+    }
     m_LastKey.Reset();
```
Comment (xhaggi) [inline xbmc/input/InputManager.cpp]: does it make sense to introduce a else here to not call reset twice?

Comment (da-anda) [inline xbmc/input/InputManager.cpp]: or copy `m_LastKey` to `key`and clear `m_LastKey` before the if-condition and use `key` in the condition;

Comment (koying) [inline xbmc/input/InputManager.cpp]: @xhaggi Yep, absolutely.
@da-anda That would always create a stack variable, even when it's not needed (i.e. 99% of the time, all the non-longpress cases)

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Comment by xhaggi** [comment]
@koying thanks

**Comment by axbmcuser** [comment]
Thank you all for fixing!

**Comment by koying** [comment]
@MartijnKaijser Did you backport to v17 yourself?

**Comment by MartijnKaijser** [comment]
Yep

**Comment by koying** [comment]
Okido, thanks :)

**Pull request merged**